packages/core-data/src/entity-provider.js

```diff
@@ -122,41 +128,104 @@ export function useEntityProp( kind, type, prop ) {
 export function __experimentalUseEntitySaving( kind, type, props ) {
 	const id = useEntityId( kind, type );
 
-	const [ isDirty, isSaving, edits ] = useSelect(
+	const [ isDirty, isSaving, _select ] = useSelect(
 		( select ) => {
 			const { getEntityRecordNonTransientEdits, isSavingEntityRecord } = select(
 				'core'
 			);
-			const _edits = getEntityRecordNonTransientEdits( kind, type, id );
-			const editKeys = Object.keys( _edits );
+			const editKeys = Object.keys(
+				getEntityRecordNonTransientEdits( kind, type, id )
+			);
 			return [
 				props ?
 					editKeys.some( ( key ) =>
 						typeof props === 'string' ? key === props : props.includes( key )
 					) :
 					editKeys.length > 0,
 				isSavingEntityRecord( kind, type, id ),
-				_edits,
+				select,
 			];
 		},
 		[ kind, type, id, props ]
 	);
 
 	const { saveEntityRecord } = useDispatch( 'core' );
 	const save = useCallback( () => {
-		let filteredEdits = edits;
+		let filteredEdits = _select( 'core' ).getEntityRecordNonTransientEdits(
```

Comment:
Because, this way we get the one from `Context`, which is not necessarily the default one.

Reply:
Interesting, should we add a comment with that to avoid a regression in the future?
> Because, this way we get the one from Context, which is not necessarily the default one.

Reply:
For this specific case, we are querying the core store. Can "select( 'core' ).getEntityRecordNonTransientEdits(..)" return different values depending on the context it is used?
If yes, could this impact plugin developers? E.g: how a plugin developer specifies which core store they want to query given that the store may change depending on context?

Reply:
>Interesting, should we add a comment with that to avoid a regression in the future?

I've added a comment.

>how a plugin developer specifies which core store they want to query given that the store may change depending on context?

That is the desired behavior. When you override the registry context, it changes for all consumers. `useSelect`, `withSelect`, `useDispatch`, etc.